src/messages.rs

```diff
@@ -202,19 +202,26 @@ pub struct GroupList {
 pub struct SignedMessage {
     /// A request or response type message.
     content: RoutingMessage,
-    /// The lists of the groups involved in routing this message, in chronological order.
-    grp_lists: Vec<GroupList>,
+    /// Nodes sending the message (those expected to sign it)
+    sending_nodes: BTreeSet<PublicId>,
```

Comment:
I wonder whether that should be a set of `SectionList`s instead: In MAID-1677, we will need to validate these public IDs using the first entry of `sec_lists`. If that first entry is the list of members of the accumulating node's section, and `sending_nodes` are section lists _signed_ by that section, that should suffice.

For validating the signatures of the message itself, you would then still use the union of these sections, as you already do.